source/atomics.tex

```diff
@@ -931,7 +931,9 @@
 \begin{note} The \tcode{memcpy} and \tcode{memcmp} semantics of the compare-and-exchange
 operations may result in failed comparisons for values that compare equal with
 \tcode{operator==} if the underlying type has padding bits, trap bits, or alternate
-representations of the same value.\end{note}
+representations of the same value. Notably, on implementations conforming to
+ISO/IEC/IEEE 60559, floating-point \tcode{-0.0} and \tcode{+0.0}
+will compare not-equal and NaNs with the same payload will compare equal.\end{note}
```

Comment:
Reading through this again, there seems to be a mismatch in how "compare equal" is used in the first and second half of the note. In the first half, we talk about "compare equal with `operator==`", but when we say "compare not-equal" and "compare equal" in the second half, we're talking about the `memcmp` semantics.

At the risk of verbosity, how about:

> \tcode{-0.0} and \tcode{+0.0} will not compare equal with \tcode{memcmp} but will compare equal with \tcode{operator==}, and NaNs with the same payload will compare equal with \tcode{memcmp} but will not compare equal with \tcode{operator==}.